### PR TITLE
[diag] add gpio diag command implementation for nrf528xx 

### PR DESCRIPTION
**Pull request by maciejbaczmanski**

### CODE REVIEW

File: src/DIAG.md [hunk 1]
```diff
@@ -1,13 +1,22 @@
-## Diagnostic module
+# Diagnostic module
 
 nRF528xx ports extend [OpenThread Diagnostics Module][diag].
 
-New commands allow for more accurate low level radio testing.
+## diag gpio
+
+Note: \<gpio\> in diag gpio commands is an integer that combines port and pin into a single, contiguous number space as follows:
```
Comment (edmont) [inline src/DIAG.md:7]:
```suggestion
Note: `\<gpio\>` in `diag gpio` commands is an integer that combines port and pin into a single, contiguous number space as follows:
```

File: src/src/diag.c [hunk 2]
```diff
@@ -406,11 +322,7 @@ static otError processCcaThreshold(otInstance *aInstance,
 }
 
 const struct PlatformDiagCommand sCommands[] = {
-    {"ccathreshold", &processCcaThreshold},
-    {"gpio", &processGpio},
-    {"id", &processID},
-    {"listen", &processListen},
-    {"temp", &processTemp},
+    {"ccathreshold", &processCcaThreshold}, {"id", &processID}, {"listen", &processListen}, {"temp", &processTemp},
```
Comment (edmont) [inline src/src/diag.c:325]: Maybe keep each option in a different line?

File: src/src/diag.c [hunk 3]
```diff
@@ -533,3 +445,90 @@ void otPlatDiagAlarmCallback(otInstance *aInstance)
...
 }
+
+otError otPlatDiagGpioSet(uint32_t aGpio, bool aValue)
+{
+    otError error = OT_ERROR_NONE;
+
+    otEXPECT_ACTION(otPlatDiagModeGet(), error = OT_ERROR_INVALID_STATE);
+    otEXPECT_ACTION(nrf_gpio_pin_present_check(aGpio), error = OT_ERROR_INVALID_ARGS);
+
+    nrf_gpio_pin_write(aGpio, (uint32_t)aValue);
```
Comment (edmont) [inline src/src/diag.c:456]: Maybe check that the pin is configured as output?

**Review by edmont — COMMENTED** [review]
LGTM, just a couple of comments.

File: src/src/diag.c [hunk 4]
```diff
@@ -533,3 +446,91 @@ void otPlatDiagAlarmCallback(otInstance *aInstance)
...
     }
 }
+
+otError otPlatDiagGpioSet(uint32_t aGpio, bool aValue)
+{
+    otError error = OT_ERROR_NONE;
+
+    otEXPECT_ACTION(otPlatDiagModeGet(), error = OT_ERROR_INVALID_STATE);
+    otEXPECT_ACTION(nrf_gpio_pin_present_check(aGpio), error = OT_ERROR_INVALID_ARGS);
+    otEXPECT_ACTION(nrf_gpio_pin_dir_get(aGpio) == NRF_GPIO_PIN_DIR_OUTPUT, error = OT_ERROR_FAILED);
```
Comment (edmont) [inline src/src/diag.c:456]: Maybe `OT_ERROR_INVALID_STATE`?

**Review by edmont — APPROVED** [review]

**Comment by maciejbaczmanski** [comment]
@jwhui could you please trigger build checks for this PR?

**Review by jwhui — APPROVED** [review]

**Pull request merged**